hs/src/Reach/Simulator/Core.hs

```diff
@@ -660,70 +648,127 @@ instance Interp LLStep where
       _ <- interp stmt
       interp step
     LLS_Stop _at -> return V_Null
-    LLS_ToConsensus _at _lct tc_send recv tc_mtime -> do
+    LLS_ToConsensus _at _lct tc_send dlr@(DLRecv {..}) tc_mtime -> do
       (g, l) <- getState
-      let mid = e_nmsgid g
-      incrMessageId
+      let actId = l_curr_actor_id l
+      phId <- getPhaseId actId
+      msgs' <- case M.lookup phId $ e_messages g of
+            Just m -> return m
+            Nothing -> do
+              let msg = NotFixedYet mempty
+              setGlobal g { e_messages = M.insert phId msg $ e_messages g }
```

Comment:
This is an ugly way to initialize for my taste. I think a `fromMaybe` that returns the default is more elegant